packages/firestore/src/local/simple_db.ts

```diff
@@ -299,9 +304,33 @@ export class SimpleDb {
         // https://developer.mozilla.org/en-US/docs/Web/API/IDBVersionChangeRequest/setVersion
         const request = indexedDB.open(this.name, this.version);
 
+        // Store information about "Clear Site Data" being detected in the
+        // "onupgradeneeded" event listener and handle it in the "onsuccess"
+        // event listener, as opposed to throwing directly from the
+        // "onupgradeneeded" event listener. Do this because throwing from the
+        // "onupgradeneeded" event listener results in a generic error being
+        // reported to the "onerror" event listener that cannot be distinguished
+        // from other errors.
```

Comment:
Thanks for the comment. I presume `onsuccess` always gets called _after_ `onupgradeneeded` ?

Reply:
Yes, that is correct: `onsuccess` is called after `onupgradeneeded` (as long as `onupgradeneeded` doesn't throw an exception)